src/api/app/models/user.rb

```diff
@@ -903,6 +900,16 @@ def update_notifications(params)
 
   private
 
+  def involved_packages_ids
+    user_maintained_packages.where.not(project_id: involved_projects_ids).pluck(:id) +
+        group_maintained_packages.where.not(project_id: involved_projects_ids).pluck(:id)
```

Comment:
Is there a way that we could keep this as an ActiveRecord::Relation object? Because currently we loads the packages, then get and id of arrays for them, and then load the packages again from those ids in `involved_packages` method. I'm not sure if its possible but maybe something like this could work?

```ruby
user_maintained_packages.group_maintained_packages
```

Since both those scopes should just be where clauses I think you can chain them together like that?